Improve display of Auth Provider settings

richard-cox opened the pull request:
- Make a common component used by all auth providers
- Component contains
  - enabled/disabled banner
  - settings which are now aligned in a table

Addresses #2274

~~Blocked on #2309~~

Notes for QA
- This effects all five types of provider (Github, SAML based, Google Auth, LDAP based and AzureAD)

mantis-toboggan-md commented:
The AuthBanner component looks good. Can you add it to the azureAD page as well? Otherwise everything looks good to me.

richard-cox commented:
@mantis-toboggan-md I've rebased and added to azure. If that looks fine let me know. Also stumbled on #2349. I've double checked and it's in master, so unrelated to this PR.
